File: frontend/web/vaadin/src/mateu/shared/apiClients/dtos/componentmetadata/CookieConsent.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";

export default interface CookieConsent extends ComponentMetadata {
    position: string
    cookieName: string
    message: string
    theme: string
    learnMore: string
    learnMoreLink: string
    dismiss: string
}

File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/CookieConsent.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import lombok.Builder;

@Builder
public record CookieConsent(
    String style,
    String cssClasses,
    CookieConsentPosition position,
    String cookieName,
    String message,
    String theme,
    String learnMore,
    String learnMoreLink, // https://www.cookiesandyou.com/
    String dismiss)
    implements Component {}
